airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java

```diff
@@ -482,7 +476,7 @@ WITH intermediate_data AS (
             )
             SELECT
             ${column_list}
-              to_json(struct(_airbyte_cast_errors AS errors)) AS _airbyte_meta,
+              to_json(struct(COALESCE((SELECT ARRAY_AGG(unnested_column_errors IGNORE NULLS) FROM UNNEST(column_errors) unnested_column_errors), []) AS errors)) AS _airbyte_meta,
```

Comment:
Can this be done without the sub-select?

Reply:
Above, since you are building the array manually ("["  + ...), I don't think you'll end up with a situation where you'll have no entries.  Is the `COALESCE` needed?

Reply:
Regarding the second question, I guess there could be an an empty array if there are no `streamColumns` - this failed a test with`For column _airbyte_meta, expected {"errors":[]} but got {"errors":null}` since `UNNEST([])` returns `NULL`. I added the `COALESCE` as a result.

Reply:
Playing around with this a bit more and not sure if there is a way to do this w/o the sub-select.. I'll merge this since it's working for now and happy to do a follow-up fix if you have a suggestion.